### ISSUE TITLE
ImportError: DLL load failed while importing KTransformersOps: The specified module was not found.

### ISSUE THREAD

**Issue by SCP12rs** [kvcache-ai/ktransformers#94]
Error during model run
![image](https://github.com/user-attachments/assets/0dd297a9-1bec-408d-a6c0-1a90e60a1aea)

**Comment by SCP12rs** [comment]
![image](https://github.com/user-attachments/assets/452f395f-7599-4a62-b15c-a6ca988a18e0)

![image](https://github.com/user-attachments/assets/0859e245-5aea-4f2a-9380-77b87d70736d)

Everything seems to be in place, but the error won't go away.

I use ktransformers-0.1.1+cu125torch24avx2-cp311-cp311-win_amd64.whl

**Comment by SCP12rs** [comment]
![image](https://github.com/user-attachments/assets/85326522-9ac9-420e-93da-b8a0850d1525)
Tried to build the project via .bat 
An error occurred.

**Comment by UnicornChan** [comment]
Do you have CUDA installed and set in the environment variables?
I see the final compilation error is due to failing to get the CUDA version.

**Comment by SCP12rs** [comment]
![image](https://github.com/user-attachments/assets/856c649b-196d-4116-a824-aa40ae49a08b)
![image](https://github.com/user-attachments/assets/015eb49c-0921-4989-a95d-c8db96b5e7cb)
Maybe I didn't patch it correctly?

**Comment by SCP12rs** [comment]
![image](https://github.com/user-attachments/assets/d8d05a64-4c1e-4cf2-b3b2-5cbec0a2ab18)

Update

**Comment by UnicornChan** [comment]
I'm very sorry about it. The Windows server I previously used for testing was tested with CUDA 12.5, but not with CUDA 12.6. The script failure might be due to the version update. I will upgrade my Windows server and re-test to verify this issue.

**Comment by SCP12rs** [comment]
Thank you. I will keep an eye on your project. Very high hopes.

**Comment by micellince** [comment]
> Thank you. I will keep an eye on your project. Very high hopes.

pip uninstall torch ....
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124 ( or 121 126.....).

**Comment by fshurra** [comment]
This might be a problem from your python. 
Please check this answer.
https://stackoverflow.com/a/71645987/29078640

I meet the same problem recently. 
the problem solved after inject the "[os.add_dll_directory](https://docs.python.org/3.8/library/os.html#os.add_dll_directory)" to the source code.

This problem might coming not using python inside conda env. 
As I will have no surprise if the conda env have patched the default manner for searching .dll 
(just like what they have done on searching .so)

**Comment by elijahxb** [comment]
![Image](https://github.com/user-attachments/assets/c42b8f56-f276-4c3e-82b7-c235bdb842e4)

same issue.
@fshurra could you please show us how to add_dll_directory，i mean where the cuda library path is?
for example, should i add the code like below?

`os.add_dll_directory('C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\lib\x64')`

**Comment by shijd32** [comment]
same issue:

File "D:\ProgramData\anaconda3\envs\Ktransformers\Lib\site-packages\ktransformers\util\custom_gguf.py", line 27, in <module>
    import KTransformersOps
ImportError: DLL load failed while importing KTransformersOps: 找不到指定的程序。

How can I solve this problem? Thank you very much!

**Comment by shijd32** [comment]
WHO KNOW?

**Comment by YoungITer** [comment]
> same issue:
> 
> File "D:\ProgramData\anaconda3\envs\Ktransformers\Lib\site-packages\ktransformers\util\custom_gguf.py", line 27, in import KTransformersOps ImportError: DLL load failed while importing KTransformersOps: 找不到指定的程序。
> 
> How can I solve this problem? Thank you very much!

I have the same problem.  Please help

**Comment by fshurra** [comment]
> ![Image](https://github.com/user-attachments/assets/c42b8f56-f276-4c3e-82b7-c235bdb842e4)
> 
> same issue. [@fshurra](https://github.com/fshurra) could you please show us how to add_dll_directory，i mean where the cuda library path is? for example, should i add the code like below?
> 
> `os.add_dll_directory('C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\lib\x64')`

@elijahxb 
Yes. Please add this directory in the  "site-packages\ktransformers\util\custom_gguf.py" 
before "import KTransformersOps"

This solved the problem for me.

**Comment by ErvinXie** [comment]
This issue has been automatically closed due to inactivity for 60 days.

If you believe this issue is still relevant, please feel free to reopen it with additional information or context.

Thank you for your contribution to KTransformers!

---
此 issue 因超过 60 天无活动而被自动关闭。

如果您认为此问题仍然存在，请随时重新打开并提供更多信息或上下文。

感谢您对 KTransformers 的贡献！
